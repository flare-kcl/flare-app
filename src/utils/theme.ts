import { createTheme } from '@shopify/restyle'

const palette = {
  purpleLight: '#8C6FF7',
  purplePrimary: '#5A31F4',
  purpleDark: '#3F22AB',

  greenLight: '#56DCBA',
  greenPrimary: '#0ECD9D',
  greenDark: '#0A906E',

  black: '#0B0B0B',
  white: '#F0F2F3',
}

const theme = createTheme({
  colors: {
    ...palette,
    mainBackground: palette.white,
    cardPrimaryBackground: palette.purplePrimary,
  },
  spacing: {
    s: 8,
    m: 16,
    l: 24,
    xl: 40,
  },
  breakpoints: {
    phone: 0,
    tablet: 768,
  },
  textVariants: {
    heading: {
      fontWeight: 'bold',
      fontSize: 30,
      lineHeight: 42.5,
      color: 'purplePrimary',
      alignSelf: 'center',
    },
  },
})

export type Theme = typeof theme
export default theme
