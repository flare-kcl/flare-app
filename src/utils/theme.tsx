import { ThemeProvider, createTheme } from '@shopify/restyle'

export const palette = {
  purple: '#726DA8',
  purpleLight: '#a537fd',
  greenLight: '#B2DBBF',

  greenPrimary: '#70C1B3',
  greenDark: '#0A906E',
  greenCorrect: '#2ed573',
  coral: '#F07167',
  red: '#ff4757',
  yellow: '#FFBF46',
  darkGrey: '#212529',
  lightGrey: '#bdc3c7',
  black: '#0B0B0B',
  white: '#F0F2F3',
}

export const theme = createTheme({
  colors: {
    ...palette,
    mainBackground: palette.white,
    cardPrimaryBackground: palette.purple,
  },
  spacing: {
    0: 0,
    1: 4,
    2: 8,
    3: 12,
    4: 16,
    5: 20,
    6: 24,
    8: 32,
    10: 40,
    12: 48,
    14: 60,
  },
  borderRadii: {
    s: 4,
    m: 8,
    round: 1000000,
  },
  breakpoints: {
    phone: 0,
    tablet: 768,
  },
  textVariants: {
    // Both used on the login scren
    heading: {
      fontFamily: 'Inter-Light',
      fontSize: 45,
      color: 'black',
      opacity: 0.7,
      alignSelf: 'center',
    },
    heading3: {
      fontFamily: 'Inter',
      fontSize: 20,
      fontWeight: '700',
    },
    caption: {
      fontFamily: 'Inter-SemiBold',
      fontSize: 20,
      color: 'black',
      opacity: 0.7,
      alignSelf: 'center',
    },
    caption2: {
      fontFamily: 'Inter-SemiBold',
      fontSize: 15,
      color: 'black',
      opacity: 0.7,
      alignSelf: 'center',
    },
    buttonLabel: {
      fontFamily: 'Inter-SemiBold',
      fontSize: 20,
      color: 'white',
    },
  },
  buttonVariants: {
    primary: {
      backgroundColor: 'purple',
      width: 300,
      height: 60,
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      borderRadius: 'm',
      alignSelf: 'flex-end',
    },
  },

  inputVariants: {
    login: {
      height: 50,
      width: 320,
      backgroundColor: 'white',
      borderColor: 'purple',
      borderWidth: 2,
      borderRadius: 'm',
      paddingLeft: 10,
    },
  },
})

// Utility HOC for use in App and Tests
export const FlareThemeProvider = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
)

export type Theme = typeof theme
export default theme
