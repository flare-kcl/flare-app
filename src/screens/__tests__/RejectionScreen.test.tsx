import { render, fireEvent } from '@testing-library/react-native'
import { FlareThemeProvider } from '@utils/theme'
import { RejectionScreen } from '@screens'

test('Text Renders Correctly', () => {
  const tree = render(
    <FlareThemeProvider>
      <RejectionScreen />
    </FlareThemeProvider>,
  ).toJSON()

  expect(tree).toMatchSnapshot()
})

test('Calls Props on button press', () => {
  // Setup mock functions
  const onExit = jest.fn()

  // Render screen
  const { queryByTestId } = render(
    <FlareThemeProvider>
      <RejectionScreen
        contactLink='mailt:flare@internet.com'
        onExit={onExit}
      />
    </FlareThemeProvider>,
  )

  // Capture all elements
  const exitButton = queryByTestId('ExitButton')

  // Check functions
  fireEvent.press(exitButton)
  expect(onExit).toBeCalledTimes(1)
})
