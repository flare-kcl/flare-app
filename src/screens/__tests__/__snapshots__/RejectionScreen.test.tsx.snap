// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Renders Correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#B2DBBF",
        "display": "flex",
        "flexDirection": "column",
        "height": "100%",
        "paddingHorizontal": 16,
        "paddingTop": 40,
        "width": "100%",
      },
    ]
  }
>
  <RCTScrollView
    height="100%"
    horizontal={false}
    width="100%"
  >
    <View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingTop": 40,
              "width": "100%",
            },
          ]
        }
      >
        <Image
          height="100%"
          maxHeight={150}
          maxWidth={240}
          opacity={0.8}
          resizeMode="contain"
          source={1}
          style={
            Array [
              Object {},
            ]
          }
          width="100%"
        />
      </View>
      <View
        style={
          Array [
            Object {
              "display": "flex",
              "paddingBottom": 24,
              "paddingTop": 40,
            },
          ]
        }
      >
        <View
          style={Object {}}
        >
          <View
            style={
              Object {
                "alignSelf": "center",
                "flexDirection": "row",
                "marginBottom": 10,
                "width": "100%",
              }
            }
          >
            <Text
              style={Object {}}
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#212529",
                      "fontFamily": "Inter",
                      "fontSize": 45,
                      "fontWeight": "bold",
                    },
                    Object {},
                  ]
                }
              >
                Thank you for your time!
              </Text>
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "flex-start",
                "marginBottom": 10,
                "marginTop": 10,
                "width": "100%",
              }
            }
          >
            <Text
              style={Object {}}
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#212529",
                      "fontFamily": "Inter",
                      "fontWeight": "500",
                    },
                    Object {},
                  ]
                }
              >
                Due to the information you have provided we have concluded that you are not an ideal participant for this research experiment.
              </Text>
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "flex-start",
                "marginBottom": 10,
                "marginTop": 10,
                "width": "100%",
              }
            }
          >
            <Text
              style={Object {}}
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#212529",
                      "fontFamily": "Inter",
                      "fontWeight": "500",
                    },
                    Object {},
                  ]
                }
              >
                If you belive you have answered any questions incorrectly then please use the button below to contact the lead reseacher of this experiment.
              </Text>
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "display": "flex",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 0,
              "flexDirection": "row",
              "opacity": 1,
              "width": "100%",
            }
          }
          testID="ExitButton"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "justifyContent": "center",
                  "width": "100%",
                },
                Array [
                  Object {
                    "backgroundColor": "#5E44AE",
                    "borderRadius": 8,
                    "height": 60,
                    "marginTop": 8,
                    "paddingVertical": 16,
                  },
                ],
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#F0F2F3",
                    "display": "flex",
                    "fontFamily": "Inter-SemiBold",
                    "fontSize": 20,
                  },
                ]
              }
            >
              Exit Experiment
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
